packages/signature-controller/src/SignatureController.test.ts

```diff
@@ -416,7 +430,41 @@ describe('SignatureController', () => {
       });
 
       it('approves message and signs', async () => {
-        await (signatureController as any)[signMethodName](messageParamsMock);
+        await new Promise(async (resolve) => {
+          signatureController.hub.once(`${rpcMethodName}:signed`, (data) => {
```

Comment:
I'm a big fan of separate unit tests for each side effect, would it be cleaner to have an independent test to verify the event is fired?

Reply:
That's true - I've created a new test called `'emits an event when the message is signed by the keyring'`